Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -176,33 +176,332 @@ struct HTTPHeader {
     let value: HTTPHeaderIndex
 }
 
+
+private extension UInt8 {
+    var isASCII: Bool {
+        return self <= 127
+    }
+
+    var maskedLowercase: UInt8 {
+        return self & 0xdf
+    }
+}
+
+
+/// Known HTTP header name. These should be created once and re-used as instances are memory intensive and "expensive" to create.
+public struct HTTPHeaderName {
+
+    /// Holds the bytes of the name + ': '
+    fileprivate let bytesWithSeperator: ContiguousArray<UInt8>
+    /// Holds the lowercased bases (masked) of the name.
+    fileprivate let lowerCaseBytes: ContiguousArray<UInt8>
```

Comment:
why isn't this a slice on `bytesWithSeparator`?

Reply:
Because the other may not be lowercase?

Reply:
@normanmaurer ah right! What about using `bytesWithSeparator.dropLast(2).lazy.map { $0.makeLowerCase() }` that should just lazily make them lowercase when needed. Sure, it's a tiny computation overhead but we're comparing them one by one anyway... Not sure we should be storing all headers twice...

Reply:
`makeLowerCase` would turn a character lower case, ie. just setting a bit in ASCII

Reply:
@normanmaurer actually your `maskedLowercase` is exactly that already

Reply:
@weissi honestly I think its fine to pre-compute all of this as this is really meant to be created once and reused. So I would trade memory for performance. WDYT ?